pkg/kubelet/gpu/nvidia/util/util.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+)
+
+const (
+	Vendor string = "NVIDIA"
+	// All NVIDIA GPUs cards should be mounted with nvidiactl and nvidia-uvm
+	NvidiaDeviceCtl string = "/dev/nvidiactl"
+	NvidiaDeviceUVM string = "/dev/nvidia-uvm"
+)
+
+// Get all the NVIDIA GPU cards' path from /dev/
+func Discovery() ([]string, error) {
+	gpulist := []string{}
+	err := filepath.Walk("/dev", func(path string, f os.FileInfo, err error) error {
+		reg := regexp.MustCompile(`^nvidia[0-9]*$`)
+		gpupath := reg.FindAllString(f.Name(), -1)
```

Comment:
Thanks for your comments, but could you be more specific which n should be 1? the return of len(reg.FindAllString(f.Name(), -1))?